plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/snowflake/TestingSnowflakeServer.java

```diff
@@ -0,0 +1,95 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg.catalog.snowflake;
+
+import io.airlift.log.Logger;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.function.Function;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static io.trino.testing.TestingProperties.requiredNonEmptySystemProperty;
+
+public class TestingSnowflakeServer
+{
+    public enum TableType
+    {
+        NATIVE, ICEBERG
+    }
+
+    private static final Logger LOG = Logger.get(TestingSnowflakeServer.class);
+
+    public static final String SNOWFLAKE_JDBC_URI = requiredNonEmptySystemProperty("testing.snowflake.catalog.account-url");
+    public static final String SNOWFLAKE_USER = requiredNonEmptySystemProperty("testing.snowflake.catalog.user");
+    public static final String SNOWFLAKE_PASSWORD = requiredNonEmptySystemProperty("testing.snowflake.catalog.password");
+    public static final String SNOWFLAKE_ROLE = requiredNonEmptySystemProperty("testing.snowflake.catalog.role");
+    public static final String SNOWFLAKE_WAREHOUSE = requiredNonEmptySystemProperty("testing.snowflake.catalog.warehouse");
+    public static final String SNOWFLAKE_TEST_DATABASE = requiredNonEmptySystemProperty("testing.snowflake.catalog.database");
+
+    public void execute(String schema, String sql)
+            throws SQLException
+    {
+        checkArgument(sql != null && !sql.isEmpty(), "sql can not be null or empty");
+        executeOnDatabaseWithResultSetOperator(schema, Optional.empty(), sql);
+    }
+
+    public boolean checkIfTableExists(TableType tableType, String schema, String tableName)
+            throws SQLException
+    {
+        Function<ResultSet, Boolean> tableExistsOperator = resultSet -> {
+            try {
+                return resultSet.next();
+            }
+            catch (SQLException e) {
+                throw new RuntimeException(e);
+            }
+        };
+        String sql = "SHOW %sTABLES LIKE '%s'".formatted(tableType == TableType.ICEBERG ? "ICEBERG " : "", tableName);
+
+        return executeOnDatabaseWithResultSetOperator(schema, Optional.of(tableExistsOperator), sql).orElseThrow();
+    }
+
+    public <T> Optional<T> executeOnDatabaseWithResultSetOperator(String schema, Optional<Function<ResultSet, T>> function, String sql)
```

Comment:
nit: This logic seems different from common JDBC driver usage. We could change the return type to `Optional<ResultSet>` and change the logic like below:
```java
            if (!statement.execute(sql)) {
                return Optional.empty();
            }
            return Optional.of(statement.getResultSet());
```

Reply:
This may not be possible as resultset is closed when statement is closed through try block.

From Java doc of Statement#close
>Note:When a Statement object is closed, its current ResultSet object, if one exists, is also closed.

Reply:
Ah, I was missing `try`. Please ignore this comment.